tools/codegen/dest/lazy_ts_lowering.py

```diff
@@ -39,3 +39,26 @@ def ts_lowering_body(f: Union[NativeFunctionsGroup, NativeFunction]) -> str:
 
     return {schema.aten_name}_out;
 """
+
+def ts_equal_method_body(f: Union[NativeFunctionsGroup, NativeFunction]) -> str:
+    func = f.functional.func if isinstance(f, NativeFunctionsGroup) else f.func
+    schema = LazyIrSchema(func)
+
+    value_comparsion = []
+    for arg in schema.positional_args:
+        if arg.is_lazy_value:
+            if isinstance(arg.lazy_type, OptionalCType):
+                value_comparsion.append(f"(!has_{arg.name} || operand(i++) == {arg.name}.value())")
```

Comment:
don't we need to check that our has_argname matches their has_argname?  e.g. if we !have_argname, but they have_argname, it is bad to reuse?